pkg/virt-handler/vm.go

```diff
@@ -888,6 +888,40 @@ func (d *VirtualMachineController) updateHotplugVolumeStatus(vmi *v1.VirtualMach
 	return volumeStatus, needsRefresh
 }
 
+func (d *VirtualMachineController) updateContainerDiskVolumeInfo(vmi *v1.VirtualMachineInstance) error {
+	findVolume := func(volumeName string) (*v1.Volume, int) {
+		for i, volume := range vmi.Spec.Volumes {
+			if volume.Name == volumeName {
+				return &volume, i
+			}
+		}
+		return nil, -1
+	}
```

Comment:
I need the volume index as well which I would need to store in another way, not sure if it reduces complexity.

Reply:
> I need the volume index as well which I would need to store in another way

You can use a map `volumeName -> {*v1.Volume, index}`

> not sure if it reduces complexity

I was referring to computational complexity. Current implementation is n^2 which is not nice in general: https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#good-to-check. With map it will be just linear.